### ISSUE TITLE
ISkRuntimeEffect changes from 3.4.0 to 6.0.0

### ISSUE THREAD

**Issue by GrooverMD** [skia4delphi/skia4delphi#301]
I am trying to run Jim McKeeth's Simple Shader Viewer.

https://github.com/jimmckeeth/SkiaSimpleShaderViewer

This demo was written with Skia4Delphi 3.4.0 but I have 6.0.0 installed, which I assume has significant changes.

This code doesn't work with 6.0.0
```delphi

//FEffect : ISkRuntimeEffect;

FEffect.ChildrenShadersByName['iImage1'] := image1.MakeShader( TSKSamplingOptions.High ); 
//undeclared indentifier 'ChildrenShadersByName'

  if( FEffect.UniformExists( 'iImage1Resolution' ) then
    case FEffect.UniformTypeByName['iImage1Resolution'] of
      TSkRuntimeEffectUniformType.Float2,
      TSkRuntimeEffectUniformType.Int2 : FEffect.SetUniform( 'iImage1Resolution', [image1.Width, image1.Height] );
      //undeclared indentifier 'SetUniform'

     TSkRuntimeEffectUniformType.Float3,
     TSkRuntimeEffectUniformType.Int3 : FEffect.SetUniform( 'iImage1Resolution', [image1.Width, image1.Height, 0] );
     //undeclared indentifier 'SetUniform'
   end;
```

And this code
```delphi
FPaint := TSkPaint.Create;
FPaint.Shader := FEffect.MakeShader;
//There is no overloaded version of 'MakeShader' that can be called with these arguments

SkAnimatedPaintBox1.Animation.Start;
```

How do I got about converting this to 6.0.0? The closest I can find to SetUniform is in the ISkRuntimeEffectBuilder interface. but I can't find any documentation on 'ISkRuntimeEffectBuilder' on how to use it, if that is the right course of action. Also what was 'ChildrenShadersByName' replaced with?

Regards,
Mark.

**Comment by viniciusfbb** [comment]
The way we did SkRuntimeEffects in v3 was a little easier to use but there was a lot of customization on our side. In the new versions, we decided to abandon our customizations and follow as close as possible to the original Google Skia API, because:
1. Documentation and demos from other Skia projects in other languages (including original Google Skia) would be valid for Skia4Delphi.
2. It would make it easier to update the library by reducing customizations at the Google Skia API level.

I suggest you take a look at the official Skia4Delphi Demo, more specifically the units:
1. https://github.com/skia4delphi/skia4delphi/blob/main/Samples/Demo/FMX/Source/Sample.Form.RuntimeEffects.pas
2. https://github.com/skia4delphi/skia4delphi/blob/main/Samples/Demo/FMX/Source/Sample.Form.Filter.pas
